lib/rules/no-restricted-exports.js

```diff
@@ -35,19 +35,51 @@ module.exports = {
                         type: "string"
                     },
                     uniqueItems: true
+                },
+                restrictDefaultExports: {
```

Comment:
Can we update the schema so that a combination `{ restrictedNamedExports: [..., "default", ...], restrictDefaultExports: { ... } }` is an invalid config? I think it would be better than silently ignoring `restrictDefaultExports` when `restrictedNamedExports` contains `"default"`.

Reply:
I think something like this would work, but please double-check:

```js
schema: [{
    type: "object",
    anyOf: [
        {
            properties: {
                restrictedNamedExports: {
                    type: "array",
                    items: {
                        type: "string"
                    },
                    uniqueItems: true
                }
            },
            additionalProperties: false
        },
        {
            properties: {
                restrictedNamedExports: {
                    type: "array",
                    items: {
                        type: "string",
                        pattern: "^(?!default$)"
                    },
                    uniqueItems: true
                },
                restrictDefaultExports: {
                    type: "object",
                    properties: {

                        // Allow/Disallow `export default foo; export default 42; export default function foo() {}` format
                        direct: {
                            type: "boolean"
                        },

                        // Allow/Disallow `export { foo as default };` declarations
                        named: {
                            type: "boolean"
                        },

                        //  Allow/Disallow `export { default } from "mod"; export { default as default } from "mod";` declarations
                        defaultFrom: {
                            type: "boolean"
                        },

                        //  Allow/Disallow `export { foo as default } from "mod";` declarations
                        namedFrom: {
                            type: "boolean"
                        },

                        //  Allow/Disallow `export * as default from "mod"`; declarations
                        namespaceFrom: {
                            type: "boolean"
                        }
                    },
                    additionalProperties: false
                }
            },
            additionalProperties: false
        }
    ]
}],
```

Reply:
This seems to work.

```js
/* eslint no-restricted-exports: ["error", { "restrictedNamedExports": ["default"], restrictDefaultExports: { direct: true, named: false, namespaceFrom: true }  }]  */
```

The above config throws the following error:
 
<img width="1259" alt="Screenshot 2023-01-17 at 6 50 08 PM" src="https://user-images.githubusercontent.com/46647141/212909507-88cd84f1-bc7e-453e-a0c5-667696a677d6.png">

Reply:
[CI failed](https://github.com/eslint/eslint/actions/runs/3939718373/jobs/6739902001), push a minor refactor in https://github.com/eslint/eslint/pull/16785/commits/037b30c60503c870fd0cdda121a41e5d7e3fa61b